students/alexander_boone/lesson06/assignments/test_mailroom.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+
+import mailroom as mail
+import os
+
+# ----- TEST FUNCTIONS -----
+
+
+def get_updated_donors_list():
+    '''Generate updated donors variables for testing purposes.'''
+    return mail.donors.keys()
+
+
+def get_text(filepath):
+    '''Get text from the specified written letter for testing.'''
+
+    with open(filepath, "r") as readfile:
+        contents = readfile.read()
+        return contents
+
+# ----- TEST CODE -----
+
+
+def test_add_donor():
+    """Test the add_donor function."""
+
+    mail.add_donor('Alexander Boone')
+    donors_list = get_updated_donors_list()
+    assert any(donor == 'Alexander Boone' for donor in donors_list)
+
+
+def test_update_donor():
+    """Test the update_donor function."""
+
+    name = 'Alexander Boone'
+    donation = 1000
+    assert mail.update_donor(name, donation) == {'Alexander Boone': [1000, 1]}
+
+
+def test_print_list():
+    """Test the print_list function."""
+
+    assert (mail.print_list() == ['Arnold Schwarzenegger', 'Lebron James',
+                                  'Elon Musk', 'Walter White',
+                                  'Gordon Ramsay', 'Alexander Boone'])
+
+
+def test_create_report():
+    """Test the create_report function for accurate rerport data creation."""
+
+    report_data = mail.create_report()
+    assert any(item == ('Arnold Schwarzenegger', 100000, 1, 100000/1)
+               for item in report_data)
+    assert any(item == ('Lebron James', 1000000, 1, 1000000/1)
+               for item in report_data)
+    assert any(item == ('Elon Musk', 2000000, 1, 2000000/1)
+               for item in report_data)
+    assert any(item == ('Walter White', 500000, 1, 500000/1)
+               for item in report_data)
+    assert any(item == ('Gordon Ramsay', 1280000, 1, 1280000/1)
+               for item in report_data)
+    assert any(item == ('Alexander Boone', 1000, 1, 1000/1)
+               for item in report_data)
```

Comment:
above is pretty inefficient.. that's is a total of 6 loops over single set of data

given that your test should also verify correct sort, you should just assert on `report_data` you expect
`assert report_data = [('Arnold Schwarzenegger', 100000, 1, 100000/1),(.. ]`